src/main/java/org/sonarlint/intellij/ui/review/ReviewSecurityHotspotDialog.kt

```diff
@@ -85,20 +91,42 @@ class ReviewSecurityHotspotDialog(
                         notification.isImportant = true
                         notification.notify(project)
 
-                        runOnUiThread(
-                            project,
-                            { close(CANCEL_EXIT_CODE) },
-                            ModalityState.stateForComponent(this@ReviewSecurityHotspotDialog.contentPane)
-                        )
+                        closeDialog(project)
                         null
                     }
             }
         }
-        centerPanel = ReviewSecurityHotspotPanel(listOfAllowedStatus, currentStatus) { changeStatusAction.isEnabled = it }
+
+        mainPanel = JBLoadingPanel(BorderLayout(), disposable)
+        reviewPanel = ReviewSecurityHotspotPanel(currentStatus) { changeStatusAction.isEnabled = canChangeStatus && it }
+        reviewPanel.preferredSize = Dimension(650, 380)
+        mainPanel.add(reviewPanel, BorderLayout.CENTER)
+        mainPanel.startLoading()
+        mainPanel.setLoadingText("Checking permissions...")
         init()
+        SonarLintUtils.getService(BackendService::class.java).checkStatusChangePermitted(connection.name, securityHotspotKey)
+            .thenAccept { checkResponse ->
+                runOnUiThread(project, ModalityState.stateForComponent(this@ReviewSecurityHotspotDialog.contentPane)) {
+                    if (!checkResponse.isPermitted) {
+                        canChangeStatus = false
+                        setErrorText(checkResponse.notPermittedReason)
+                    }
+                    reviewPanel.display(checkResponse.allowedStatuses)
+                    changeStatusAction.isEnabled = checkResponse.isPermitted
```

Comment:
By doing this, you override the callback function above saying that it should be enabled if the current status is different than the one currently applied on SQ.

I would simply change `canChangeStatus` to false or true depending on `if (!checkResponse.isPermitted)`, and remove this line, it should make it works